sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q25.sf100/explain.txt

```diff
@@ -1,53 +1,50 @@
-      +- * HashAggregate (46)
-         +- * Project (45)
-            +- * SortMergeJoin Inner (44)
-               :- * Sort (35)
-               :  +- Exchange (34)
-               :     +- * Project (33)
-               :        +- * SortMergeJoin Inner (32)
-               :           :- * Sort (23)
-               :           :  +- Exchange (22)
-               :           :     +- * Project (21)
-               :           :        +- * SortMergeJoin Inner (20)
-               :           :           :- * Sort (14)
-               :           :           :  +- Exchange (13)
-               :           :           :     +- * Project (12)
-               :           :           :        +- * BroadcastHashJoin Inner BuildRight (11)
-               :           :           :           :- * Project (6)
-               :           :           :           :  +- * BroadcastHashJoin Inner BuildRight (5)
-               :           :           :           :     :- * Filter (3)
-               :           :           :           :     :  +- * ColumnarToRow (2)
-               :           :           :           :     :     +- Scan parquet default.store_sales (1)
-               :           :           :           :     +- ReusedExchange (4)
-               :           :           :           +- BroadcastExchange (10)
-               :           :           :              +- * Filter (9)
-               :           :           :                 +- * ColumnarToRow (8)
-               :           :           :                    +- Scan parquet default.store (7)
-               :           :           +- * Sort (19)
-               :           :              +- Exchange (18)
-               :           :                 +- * Filter (17)
-               :           :                    +- * ColumnarToRow (16)
-               :           :                       +- Scan parquet default.item (15)
-               :           +- * Sort (31)
-               :              +- Exchange (30)
-               :                 +- * Project (29)
-               :                    +- * BroadcastHashJoin Inner BuildRight (28)
-               :                       :- * Filter (26)
-               :                       :  +- * ColumnarToRow (25)
-               :                       :     +- Scan parquet default.store_returns (24)
-               :                       +- ReusedExchange (27)
-               +- * Sort (43)
-                  +- Exchange (42)
-                     +- * Project (41)
-                        +- * BroadcastHashJoin Inner BuildRight (40)
-                           :- * Filter (38)
-                           :  +- * ColumnarToRow (37)
-                           :     +- Scan parquet default.catalog_sales (36)
-                           +- ReusedExchange (39)
+TakeOrderedAndProject (46)
++- * HashAggregate (45)
+   +- Exchange (44)
+      +- * HashAggregate (43)
+         +- * Project (42)
+            +- * SortMergeJoin Inner (41)
```

Comment:
comparing to before: `Exchange (34)` is no longer required since join keys are:
left: [sr_customer_sk#19, **sr_item_sk#18**] 
right: [2]: [cs_bill_customer_sk#26, **cs_item_sk#27**]

and partitioning are:
left: (hashpartitioning(ss_item_sk#1, 5) or hashpartitioning(i_item_sk#14, 5)) or **hashpartitioning(sr_item_sk#18, 5)**
right: **hashpartitioning(cs_item_sk#27, 5)**